### PR TITLE
[Merged by Bors] - feat: the category of short complexes has limits

### PR DESCRIPTION
**Pull request by joelriou**
---
- [x] depends on: #6245

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/Limits.lean [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  ⟨fun _ _ _ => inferInstance⟩
+
+end
+
+section
+
+variable [HasLimitsOfShape WalkingCospan C]
+
+instance preservesMonomorphisms_π₁ :
+  Functor.PreservesMonomorphisms (π₁ : _ ⥤ C) :=
```
Comment (kbuzzard) [inline Mathlib/Algebra/Homology/ShortComplex/Limits.lean:139]:
```suggestion
    Functor.PreservesMonomorphisms (π₁ : _ ⥤ C) :=
```
and similarly for the next two instances

**Review by kbuzzard — APPROVED** [review]
I was a little nervous when I saw your definition of `s` but you can prove things about it so it must be OK :-)

**Review by adamtopaz — COMMENTED** [review]
LGTM! Just one small comment, but I would be happy with this either way.

bors d+

File: Mathlib/Algebra/Homology/ShortComplex/Limits.lean [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    have eq₂ := h₂.fac (π₂.mapCone s)
+    have eq₃ := h₃.fac (π₃.mapCone s)
+    have eq₁₂ := fun j => (c.π.app j).comm₁₂
+    have eq₁₂' := fun j => (s.π.app j).comm₁₂
+    have eq₂₃ := fun j => (c.π.app j).comm₂₃
+    have eq₂₃' := fun j => (s.π.app j).comm₂₃
+    dsimp at eq₁ eq₂ eq₃ eq₁₂ eq₁₂' eq₂₃ eq₂₃'
+    exact Hom.mk (h₁.lift (π₁.mapCone s)) (h₂.lift (π₂.mapCone s)) (h₃.lift (π₃.mapCone s))
+      (h₂.hom_ext (fun j => by dsimp; rw [assoc, assoc, ← eq₁₂, reassoc_of% eq₁, eq₂, eq₁₂']))
+      (h₃.hom_ext (fun j => by dsimp; rw [assoc, assoc, ← eq₂₃, reassoc_of% eq₂, eq₃, eq₂₃']))
```
[the remark this answers is not in the document]
Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/Limits.lean]: I do not think it changes much in that case, but I have now defined the fields of the structure.

**Comment by joelriou** [comment]
bors merge

**Pull request closed without merging**